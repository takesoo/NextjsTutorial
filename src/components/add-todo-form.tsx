import { Button, Input } from "@chakra-ui/react";

type AddTodoFormProps = {
	newTodoTitle: string;
	setNewTodoTitle: (title: string) => void;
	addTodo: () => void;
};
export default function AddTodoForm({
	newTodoTitle,
	setNewTodoTitle,
	addTodo,
}: AddTodoFormProps): JSX.Element {
	return (
		<>
			<Input
				placeholder="New Todo"
				value={newTodoTitle}
				onChange={(e) => setNewTodoTitle(e.target.value)}
			/>
			<Button onClick={addTodo} colorScheme="blue" mb={4}>
				Add Todo
			</Button>
		</>
	);
}
