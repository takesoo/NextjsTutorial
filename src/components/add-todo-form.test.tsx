import { fireEvent, render, screen } from "@testing-library/react";
import { AddTodoForm } from "./add-todo-form";

describe("AddTodoForm", () => {
  const mockSetTodoTitle = jest.fn();
  const mockAddTodo = jest.fn();
  it("renders input and button", () => {
    render(
      <AddTodoForm
        newTodoTitle=""
        setNewTodoTitle={mockSetTodoTitle}
        addTodo={mockAddTodo}
      />,
    );
    expect(screen.getByPlaceholderText("New Todo")).toBeInTheDocument();
    expect(
      screen.getByRole("button", { name: /add todo/i }),
    ).toBeInTheDocument();
  });

  it("calls addTodo when the button is clicked", () => {
    render(
      <AddTodoForm
        newTodoTitle="Test Todo"
        setNewTodoTitle={mockSetTodoTitle}
        addTodo={mockAddTodo}
      />,
    );
    const button = screen.getByRole("button", { name: /add todo/i });
    fireEvent.click(button);
    expect(mockAddTodo).toHaveBeenCalled();
  });

  it("calls setNewTodoTitle on input change", () => {
    render(
      <AddTodoForm
        newTodoTitle=""
        setNewTodoTitle={mockSetTodoTitle}
        addTodo={mockAddTodo}
      />,
    );
    const input = screen.getByPlaceholderText("New Todo");
    fireEvent.change(input, { target: { value: "New Task" } });
    expect(mockSetTodoTitle).toHaveBeenCalledWith("New Task");
  });
});
