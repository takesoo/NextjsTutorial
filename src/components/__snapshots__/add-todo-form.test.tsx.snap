// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodoForm component matches snapshot 1`] = `
<div>
  <input
    class="chakra-input css-0"
    placeholder="New Todo"
    value=""
  />
  <button
    class="chakra-button css-eaw8jh"
    disabled=""
    type="button"
  >
    Add Todo
  </button>
</div>
`;
